File: android/app/src/main/java/com/example/jamboree/theme/JamboreeTheme.kt
package com.example.jamboree.theme

import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.darkColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/**
 * Brand tokens for the Jamboree UI redesign — black canvas, neon-magenta
 * accent, Roboto + monospaced labels. [Ub] is the single source of truth
 * used by every redesigned shared screen and atom (`JamboreeKit.kt`).
 *
 * Per-game screens are not wrapped in this theme — they keep the default
 * MaterialTheme from [MainActivity].
 */
object Ub {
    val Accent = Color(0xFFFF2E88)        // primary action, host pip, focus
    val OnAccent = Color(0xFF2A0010)      // ink on magenta
    val Foreground = Color.White          // body text, guest pips
    val Background = Color(0xFF000000)    // app bg
    val Canvas = Color(0xFF0A0A0A)        // screen canvas
    val Surface = Color(0xFF141416)       // card / list-row bg
    val SurfaceHi = Color(0xFF1C1C1F)     // hover / pressed / glyph tile
    val Line = Color.White.copy(alpha = 0.08f)        // hairline dividers
    val LineStrong = Color.White.copy(alpha = 0.14f)  // button outline
    val Muted = Color.White.copy(alpha = 0.58f)       // secondary text
    val Faint = Color.White.copy(alpha = 0.38f)       // tertiary / metadata
    val AccentSoft = Color(0xFFFF2E88).copy(alpha = 0.14f)  // selected-row fill
    val AccentLine = Color(0xFFFF2E88).copy(alpha = 0.45f)  // selected-row border
    val Online = Color(0xFF3DDC84)        // connected pip

    object Radius {
        val chip = 8.dp
        val button = 12.dp
        val row = 14.dp
        val card = 16.dp
        val panel = 18.dp
        val hero = 22.dp
    }
}

private val JamboreeColorScheme = darkColorScheme(
    primary = Ub.Accent,
    onPrimary = Ub.OnAccent,
    secondary = Ub.Accent,
    onSecondary = Ub.OnAccent,
    background = Ub.Canvas,
    onBackground = Ub.Foreground,
    surface = Ub.Surface,
    onSurface = Ub.Foreground,
    surfaceVariant = Ub.SurfaceHi,
    onSurfaceVariant = Ub.Muted,
    // Drift elevated surfaces magenta to match the mock.
    surfaceTint = Ub.Accent,
    outline = Ub.LineStrong,
    error = Ub.Accent,
)

@Composable
fun JamboreeTheme(content: @Composable () -> Unit) {
    MaterialTheme(colorScheme = JamboreeColorScheme) {
        Surface(color = Ub.Canvas, contentColor = Ub.Foreground) {
            content()
        }
    }
}

File: ios/Jamboree/App/JamboreeTheme.swift
import SwiftUI

/// Brand tokens for the Ubapp UI redesign — black canvas, neon-magenta
/// accent, system sans + monospaced labels. These are the single source of
/// truth used by every redesigned screen and atom (`UbappKit.swift`).
///
/// Per-game views may still set their own palettes, but the shared chrome
/// (menu, lobby, join, settings) and the new atoms read from here. Apply
/// `ubappChrome()` on each non-game root rather than the enclosing
/// `NavigationStack`, so pushed game screens still render normally.
enum UbappTheme {
    // Core brand
    static let accent = Color(hex: 0xFF2E88)        // primary action, host pip, focus
    static let onAccent = Color(hex: 0x2A0010)      // ink on magenta
    static let foreground = Color.white             // body text, guest pips

    // Backgrounds
    static let background = Color.black              // #000000 app bg
    static let canvas = Color(hex: 0x0A0A0A)         // screen canvas
    static let surface = Color(hex: 0x141416)        // card / list-row bg
    static let surfaceHi = Color(hex: 0x1C1C1F)      // hover / pressed / glyph tile

    // Lines & text tiers
    static let line = Color.white.opacity(0.08)         // hairline dividers
    static let lineStrong = Color.white.opacity(0.14)   // button outline
    static let muted = Color.white.opacity(0.58)        // secondary text
    static let faint = Color.white.opacity(0.38)        // tertiary / metadata

    // Accent fills
    static let accentSoft = Color(hex: 0xFF2E88).opacity(0.14)  // selected-row fill
    static let accentLine = Color(hex: 0xFF2E88).opacity(0.45)  // selected-row border

    // Status
    static let online = Color(hex: 0x3DDC84)         // connected pip
}

/// Corner radii from the spec: chips 8, buttons 12, rows 14, cards 16,
/// panels 18, hero blocks 22.
enum UbappRadius {
    static let chip: CGFloat = 8
    static let button: CGFloat = 12
    static let row: CGFloat = 14
    static let card: CGFloat = 16
    static let panel: CGFloat = 18
    static let hero: CGFloat = 22
}

extension Color {
    /// 0xRRGGBB literal initializer for brand tokens.
    init(hex: UInt32) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255,
            green: Double((hex >> 8) & 0xFF) / 255,
            blue: Double(hex & 0xFF) / 255,
            opacity: 1,
        )
    }
}

private struct UbappChromeModifier: ViewModifier {
    func body(content: Content) -> some View {
        content
            .background(UbappTheme.canvas.ignoresSafeArea())
            .tint(UbappTheme.accent)
            .foregroundStyle(UbappTheme.foreground)
            .toolbarBackground(UbappTheme.canvas, for: .navigationBar)
            .toolbarBackground(.visible, for: .navigationBar)
            .toolbarColorScheme(.dark, for: .navigationBar)
    }
}

extension View {
    func ubappChrome() -> some View { modifier(UbappChromeModifier()) }
}
